index.js

```diff
@@ -112,7 +112,7 @@ function compression (options) {
 
       // write Buffer for Node.js 0.8
       return chunk
-        ? stream.end(toBuffer(chunk, encoding))
+        ? stream.end(toBuffer(chunk, encoding), cb || encoding)
```

Comment:
I think a test case needs to be added for this case, as I removed this change and tests still passed.